x/bonds/handler.go

```diff
@@ -333,32 +336,64 @@ func handleMsgSetNextAlpha(ctx sdk.Context, keeper keeper.Keeper, msg types.MsgS
 	// Get current parameters
 	paramsMap := bond.FunctionParameters.AsMap()
 
-	// Check 1 (newAlpha != alpha)
-	if newAlpha.Equal(paramsMap["alpha"]) {
+	// Check (newPublicAlpha != publicAlpha)
+	if newPublicAlpha.Equal(paramsMap["publicAlpha"]) {
 		return nil, sdkerrors.Wrap(types.ErrInvalidAlpha,
-			"cannot change alpha to the current value of alpha")
+			"cannot change public alpha to the current value of public alpha")
 	}
-	// Check 2 (I > C * alpha)
-	if paramsMap["I0"].LTE(newAlpha.MulInt(C)) {
+
+	// Calculate scaled delta public alpha, to calculate new system alpha
+	prevPublicAlpha := paramsMap["publicAlpha"]
+	deltaPublicAlpha := newPublicAlpha.Sub(prevPublicAlpha)
+	temp, err := types.ApproxPower(
+		prevPublicAlpha.Mul(sdk.OneDec().Sub(types.StartingPublicAlpha)),
+		sdk.MustNewDecFromStr("2"))
+	if err != nil {
+		return nil, err
+	}
+	scaledDeltaPublicAlpha := deltaPublicAlpha.Mul(temp)
+
+	// Calculate new system alpha
+	prevSystemAlpha := paramsMap["systemAlpha"]
+	var newSystemAlpha sdk.Dec
+	if deltaPublicAlpha.IsPositive() {
+		// 1 - (1 - scaled_delta_public_alpha) * (1 - previous_alpha)
+		temp1 := sdk.OneDec().Sub(scaledDeltaPublicAlpha)
+		temp2 := sdk.OneDec().Sub(prevSystemAlpha)
+		newSystemAlpha = sdk.OneDec().Sub(temp1.Mul(temp2))
+	} else {
+		// (1 - scaled_delta_public_alpha) * (previous_alpha)
+		temp1 := sdk.OneDec().Sub(scaledDeltaPublicAlpha)
+		temp2 := prevSystemAlpha
+		newSystemAlpha = temp1.Mul(temp2)
+	}
+
+	// Check 1 (newSystemAlpha != systemAlpha)
+	if newSystemAlpha.Equal(paramsMap["systemAlpha"]) {
```

Comment:
Could we have used `prevSystemAlpha` here?

Reply:
Yep, thanks!

Reply:
Applied the same changes in `x/bonds/internal/keeper/batch.go`